### DISCUSSION TITLE
Audio hisses on wake SP7 i7

### DISCUSSION THREAD

**Discussion by okadrian** [badstorm/surface-pro-7-opencore#126] [Q&A]
I get a hiss from the speakers when waking the laptop which continues until I trigger some other sound. At which point there is a loud click from the speakers, then the audio system behaves normally until the sleep wake hiss cycle. Is this just me?

Edit: I'm using Monterey 12.5.1

**Comment by xealstorm** [comment]
I have the same on my Monterey i7 SP7.

**Comment by WinkeyLin** [comment]
The same on i5 SP7 Monterey 12.6.5 with the latest release of EFI.

**Comment by szsong** [comment]
same here, SP7 i7, running Ventura with latest release of EFI

**Comment by f-1-0** [comment]
I found that setting bit depth to 24 and sample rate to highest available in audio-midi-setup made it disappear. (takes re-boot)

**Comment by badstorm** [comment]
Since the conversation is dated I am closing the issue and moving it as a conversation.

**Reply by f-1-0** [reply]
btw re-setting the bit depth etc as suggested in december did not solve this.
Our layout-ID seems to be quickly copied from layout ID21, as it retains the typo in `<key>PathMapID</key><integer>212</integer>` (should be 35 or 21 respectively)
I also suspect the hissing has something to do with these values in platforms35.xml `<key>MuteInputAmp</key><false/>`

```
<key>PublishMute</key>
<false/>
<key>PublishVolume</key>
<true/>
<key>VolumeInputAmp</key>
<false/>
```


However I failed to properly built a modified AppleALC.kext to try it out.
Perhaps original maker of this layout ID @jackjack1-su can help?

**Comment by dmedina0930** [comment]
I dont know if this is relevent anymore but i had to use layout-id 35 for my surface laptop 3 which also uses ALC274, I have made a custom id 108 and i have fixed the hissing on wake by adding wake/reinitialization verbs to WakeConfig Data in PinConfigs.kext Plist 
<key>WakeConfigData</key>
<data>AbcBAQG3B0ABtwwCAbNgAAA2AzwCFwEAAhcHwAIXDAICFgOA</data>
As of now I still have the issue of the static/hissing on the speakers when I plug in headphones (Headphones have audio)

Could a Surface Pro 7 user test this?

I have attached my AppleALC.kext. just alcid=108 or in device properties (if setup) layout-id = 108
[AppleALC.kext.zip](https://github.com/user-attachments/files/30726568/AppleALC.kext.zip)
I am on sonoma 14.8.5 
Sleep still hasn't been fixed 😭🙏

**Reply by f-1-0** [reply]
Just tested your adapted kext and it works for me!
The hiss is gone after wake from S4 sleep! It can still be triggered when opening system settings and selecting the input/internal mic. (on Sequoia 15.3)

**Reply by dmedina0930** [reply]
That's good to hear!
I replicated that same issue with the microphone in settings. 
Now I just have to figure out the mic and the jack.

**Reply by dmedina0930** [reply]
Here is the latest one I still havent fixed the hissing at least for the surface laptop 3 
I added mute/unmute stuff for the DACs and ADCs to the ConfigData but seems to have made no difference at least for the surface laptop 3 
Im going to take a break from it for my sanity if anyone else wants to take a crack at it.
[LatestAppleALC.kext.zip](https://github.com/user-attachments/files/30754753/LatestAppleALC.kext.zip)
